tech_reports/LLMs/llms.md

```diff
@@ -86,7 +87,231 @@ Authors:
   - Accuracy tests
   - Debugging PCC issues
 ### 4.6 Performance Analysis
-  - Performance tooling, tracy
+
+Think of ttnn performance as having five components:
+
+![Performance components overview](images/4.6-overview.png)
+
+1. **Main python thread** - this is your code that executes the ttnn calls and does other bits of logic in between. The speed of this thread determines the speed at which python calls are dispatched to the API. You are in control of any overheads here. When counting in microseconds python is probably slower than you think.
+2. **Host API** - most of your ttnn calls will be immediately dispatched onto multiple C++ threads for further processing before anything is done to the hardware. You are generally not in control of any overheads in this part of the stack.
+3. **Host-device communications** - data is heavy and moving it is to be avoided. PCIe bandwidth and latency is not negligible at the kinds of speeds we want to run models at. In addition, Tenstorrent converts most data into tiles of 32x32 elements for faster processing. Tilizing and untilizing data takes time and should be performed on-device wherever possible, but you will have to specify that as we’ll see in this section.
+4. **Device dispatch** - we can measure the gap between one op finishing and the next starting. At time of writing the lower limit of this is single-digit microseconds and there is work underway to reduce it to zero. However, for various reasons you might see much higher dispatch times, most notably if there are a lot of runtime arguments to a function or if something else is happening in between calls.
```

Comment:
I think this work has been partially pushed to main already. So, maybe mention that dispatch will be overlapped with previous op execution whenever possible.

Reply:
I don't see any zero dispatch times, they're all 4us with tracing enabled on main atm.